buildSrc/src/main/kotlin/org/gradle/plugins/compile/GradleCompilePlugin.kt

```diff
@@ -0,0 +1,61 @@
+package org.gradle.plugins.compile
+
+import accessors.java
+import org.gradle.api.JavaVersion
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.internal.project.ProjectInternal
+import org.gradle.api.tasks.compile.AbstractCompile
+import org.gradle.api.tasks.compile.CompileOptions
+import org.gradle.api.tasks.compile.GroovyCompile
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.jvm.toolchain.internal.JavaInstallationProbe
+import org.gradle.kotlin.dsl.the
+import org.gradle.kotlin.dsl.withType
+
+
+open class GradleCompilePlugin : Plugin<Project> {
+    override fun apply(project: Project) {
```

Comment:
:o: You can use `project.run {}` in order to make the body of the apply function closer to script DSL.

```kotlin
override fun apply(project: Project): Unit = project.run {
    if(rootProject == project) {
        // ...
    }
    // ...
    tasks.withType<JavaCompile> {
        // ...
    }
    // ...
}
```

Reply:
Done.

Reply:
:o: You can get rid of one more braces level, see my snippet above.

Reply:
Thanks, done!